configmap/filter.go

```diff
@@ -42,3 +45,17 @@ func TypeFilter(ts ...interface{}) func(func(string, interface{})) func(string,
 		}
 	}
 }
+
+// ValidateType checks the type of the constructor it evaluates
+// for expected returned values.
+//
+// The expectation is for the constructor to return two values
+// with the second value being of type error
+func ValidateType(cType reflect.Type) error {
```

Comment:
You can probably pull more validation into this method - there's still some duplication.

I'd also rename the method to be more specific ie. `ValidateConstructor`

Reply:
Can we just have it take the constructor itself?

Reply:
yeah, if you pass the constructor here, you can just check in one place that it's a Func and the expected signature.

Reply:
yeah overlooked the lines prior. fixed.